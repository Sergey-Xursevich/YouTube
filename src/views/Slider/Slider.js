export default class Slider {
  render() {
    const section = document.createElement('div');
    const container = document.createElement('div');
    const prevLeft = document.createElement('button');
    const prevRight = document.createElement('button');
    const divButton = document.createElement('div');
    const circle = document.createElement('div');
    const slider = document.createElement('div');

    document.body.appendChild(section);
    section.classList.add('section-slider');

    section.appendChild(container);
    container.classList.add('slider-container');

    document.body.appendChild(divButton);
    divButton.classList.add('slider-container__button');

    document.querySelector('.slider-container').appendChild(slider);
    slider.classList.add('slider');

    divButton.appendChild(prevLeft);
    prevLeft.innerHTML = '<i class="fa fa-chevron-left fa-2x"></i>';
    prevLeft.classList.add('prev');

    divButton.appendChild(circle);
    circle.classList.add('round');
    circle.innerHTML = '1';

    divButton.appendChild(prevRight);
    prevRight.innerHTML = '<i class="fa fa-chevron-right fa-2x"></i>';
    prevRight.classList.add('next');
  }

  static delete() {
    const slider = document.querySelector('.slider');
    const child = document.querySelectorAll('.slide');

    for (let i = 0; i < child.length; i += 1) {
      slider.removeChild(child[i]);
    }
  }
}
