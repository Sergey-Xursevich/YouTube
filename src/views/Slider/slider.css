.section-slider {
  display: grid;
  justify-items: center;
  align-items: center;
  padding: 20px 0 0 0;
}

.slider-container {
  position: relative;
}

.slider {
  display: flex;
  width: 82vw;
  overflow-x: auto;
  overflow-y: hidden;
  scroll-snap-type: x mandatory;
  -webkit-overflow-scrolling: touch;
  scroll-behavior: smooth;
  user-select: none;
}

.slider:hover {
  cursor: pointer;
}

.slider-container__button {
  display: flex;
  justify-content: center;
  opacity: 0;
}

.slider-container__button-opacity {
  opacity: 1;
}

.slider-container__button button {
  position: absolute;
  border: none;
  background-color: rgba(255,127,80,0.2);
  color: #fff;
  cursor: pointer;
  padding: 10px 15px;
  border-radius: 50%;
  outline: none;
  opacity: 0.5;
  transition: all ease-in-out 0.5s;
}

.round {
  position: absolute;
  border: none;
  background-color: rgb(255,127,80);
  color: #fff;
  cursor: pointer;
  padding: 10px 17px;
  border-radius: 50%;
  outline: none;
  transition: all ease-in-out 0.5s;
  margin: 25px 0;
  font-size: 24px;
}

.next {
  margin: 25px 80px;
}

.prev {
  margin: 25px -80px;
}

.slider-container__button:hover button {
  opacity: 1;
}

.slider-container__button button:hover {
  background-color: coral;
}

.slider::-webkit-scrollbar {
  display: none;
}

@media (pointer:coarse) {
  .slide {
    width: 295px;
  }

  .image {
    height: 220px;
  }

  .prev, .next {
    display: none;
  }
}

@media (pointer:coarse) and (orientation: landscape) {
  .slide {
    width: 260px;
    overflow: hidden;
    margin: 10px 7px;
  }

  .image {
    height: 220px;
  }

  .prev, .next {
    display: none;
  }
}
