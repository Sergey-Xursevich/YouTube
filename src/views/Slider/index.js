import Slider from './Slider';

import './slider.css';

export default Slider;
