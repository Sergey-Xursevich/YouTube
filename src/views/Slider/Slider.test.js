import Slider from './Slider';

describe('Slider render', () => {
  it('Should be an instance of Function', () => {
    expect(Slider.prototype.render).toBeInstanceOf(Function);
  });
});

describe('Slider delete', () => {
  it('Should be an instance of Function', () => {
    expect(Slider.delete).toBeInstanceOf(Function);
  });
});

describe('Should be an render correctly', () => {
  Slider.prototype.render();
  expect(document.body.innerHTML).toMatchSnapshot();
});

describe('Should be an delete correctly', () => {
  Slider.delete();
  expect(document.body.innerHTML).toMatchSnapshot();
});
