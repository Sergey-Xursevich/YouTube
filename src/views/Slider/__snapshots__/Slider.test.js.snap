// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"<div class=\\"section-slider\\"><div class=\\"slider-container\\"><div class=\\"slider\\"></div></div></div><div class=\\"slider-container__button\\"><button class=\\"prev\\"><i class=\\"fa fa-chevron-left fa-2x\\"></i></button><div class=\\"round\\">1</div><button class=\\"next\\"><i class=\\"fa fa-chevron-right fa-2x\\"></i></button></div>"`;

exports[` 2`] = `"<div class=\\"section-slider\\"><div class=\\"slider-container\\"><div class=\\"slider\\"></div></div></div><div class=\\"slider-container__button\\"><button class=\\"prev\\"><i class=\\"fa fa-chevron-left fa-2x\\"></i></button><div class=\\"round\\">1</div><button class=\\"next\\"><i class=\\"fa fa-chevron-right fa-2x\\"></i></button></div>"`;
